src/app/components/audio-recordings/pages/analysis-results/analyses-results.component.ts

```diff
@@ -0,0 +1,185 @@
+      name: "system",
+      id: "system",
+    });
+  }
+
+  public ngOnInit() {
+    this.getItems()
+      .then((items: AnalysisJobItemResult[]) => {
+        this.rows = items;
+        this.updateRows();
+      });
+  }
+
+  public updateRows = (): Observable<AnalysisJobItemResult[]> =>
+    this.rows$ = this.getRows();
+
+  public getRows() {
+    const data = new Observable<AnalysisJobItemResult[]>((observer) => {
+      observer.next(this.rows);
+    });
+    return data;
+  }
+
+  /**
+   * Fetches the `AnalysisJobItemResult` model from the baw-api
+   *
+   * @param model The id of the model that is requested from the baw-api
+   * @returns The complete `AnalysisJobItemResult` model of the requested item. If the item is not defined, the root path will be returned
+   */
+  public async getItems(model?: AnalysisJobItemResult): Promise<AnalysisJobItemResult[]> {
+    return new Promise((resolveTo) => {
+      const analysisJobId = this.analysisJob;
+      const response = lastValueFrom(this.api
+        .list(analysisJobId, this.audioRecording.id, model)
+        .pipe(first())
+        .pipe(
+          takeUntil(this.unsubscribe)
+        )
+      );
+
+      resolveTo(response);
+    });
+  }
+
+  // this method needs to be refactored
+  public toggleRow(model: AnalysisJobItemResult) {
+    this.getModelChildren(model.children ?? [], model).then((childItems) => {
+      this.addParentInformation(childItems, model);
+
+      if (model.open) {
+        // the relative path for all sub folders should include the folders name & two back slashes before and after the name
+        // this prevents false positives e.g. if the folder name is contained within another folders name
+        this.rows = this.rows.filter(item =>
+          // only the root item will not have a parent item, in this case, we can quickly determine that all folders need to be closed
+          !isInstantiated(item.parentItem) ||
+          !item.parentItem.path.includes(`/${model.name}/`) ||
+          item.name === model.name
+        );
+      } else {
+        this.rows.splice(this.rows.indexOf(model) + 1, 0, ...childItems);
+      }
+
+      model.open = !model.open;
+      this.updateRows();
+    });
+  }
+
+  // not all analysisJobItemResult models from the baw-api contain a path attribute therefore, we can not know the location of the item
+  // therefore, by adding information about the parent object, we can easily derive the relative path and hierarchy of a sub item
+  private addParentInformation(models: AnalysisJobItemResult[], parent: AnalysisJobItemResult) {
+    models.forEach(analysisJobItemResultItem => {
+      analysisJobItemResultItem.parentItem = parent;
+    });
+  }
+
+  /**
+   * Sub directories of `AnalysisJobItemResults` do not contain all the information needed e.g. How many child items the sub directory has.
+   * Therefore, we need to evaluate the child sub directory items explicitly and use their `AnalysisJobItemResult` model.
```

Comment:
it looks like this method is doing n+1 requests for every directory. This shouldn't be happening

Reply:
The counts are needed to display the number of children in a directory (the little number 3 in the example below).

![image](https://user-images.githubusercontent.com/33742269/207994270-be5aafb2-b206-4a76-ab17-ce972eb63cbc.png)

If we were to remove this requirement, it would drastically reduce the complexity of the code, and resolve about half of the comments on this review.

Reply:
was this ever a stated requirement? Consider it removed

But the solution here is to ask for an API patch, not write hugely inefficient code